packages/test/test-utils/src/testFluidObjectInternal.ts

```diff
@@ -0,0 +1,91 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	IFluidHandle,
+	IRequest,
+	IResponse,
+	type IFluidLoadable,
+} from "@fluidframework/core-interfaces";
+import { fail } from "@fluidframework/core-utils/internal";
+import { FluidObjectHandle } from "@fluidframework/datastore/internal";
+import {
+	IChannelFactory,
+	IFluidDataStoreRuntime,
+	type IChannel,
+} from "@fluidframework/datastore-definitions/internal";
+import {
+	IFluidDataStoreChannel,
+	IFluidDataStoreContext,
+} from "@fluidframework/runtime-definitions/internal";
+import { create404Response } from "@fluidframework/runtime-utils/internal";
+import type { ISharedObject } from "@fluidframework/shared-object-base/internal";
+
+/**
+ * A test Fluid object that will create a shared object for each key-value pair in the factoryEntries passed to load.
+ * The shared objects can be retrieved by passing the key of the entry to {@link TestFluidObjectInternal.getInitialSharedObject}.
+ * It exposes the IFluidDataStoreContext and IFluidDataStoreRuntime.
+ * @remarks
+ * This is a simplified (does not use SharedMap) alternative to {@link TestFluidObject} which does not implement the external facing {@link ITestFluidObject} interface.
+ * @internal
+ */
+export class TestFluidObjectInternal implements IFluidLoadable {
+	public get IFluidLoadable() {
+		return this;
+	}
+
+	public readonly handle: IFluidHandle<this>;
+	private initializationPromise: Promise<void> | undefined;
+
+	/**
+	 * Creates a new TestFluidObjectInternal.
+	 * @param runtime - The data store runtime.
+	 * @param context - The data store context.
+	 * @param factoryEntries - A list of id to IChannelFactory mapping. For each item in the list,
+	 * a shared object is created which can be retrieved by calling {@link TestFluidObjectInternal.getInitialSharedObject} with the id;
+	 * @param initialSharedObjectsFactories - A collection of ids (which can be passed to {@link TestFluidObjectInternal.getInitialSharedObject})
+	 * and the corresponding factories to use to create the shared objects during initialization.
+	 */
+	constructor(
+		public readonly runtime: IFluidDataStoreRuntime,
+		public readonly channel: IFluidDataStoreChannel,
+		public readonly context: IFluidDataStoreContext,
+		private readonly initialSharedObjectsFactories: ReadonlyMap<
```

Comment:
Rename similar property in `TestFluidObjectFactory` and `TestFluidObject` too.

Reply:
I guess those are not part of the stable API, so such a rename is possible. I'll do it.